library/datasort.c

```diff
@@ -576,6 +651,21 @@ static int datasort_split(struct datasort_cfg *dcfg)
 	for (n = 0; n < dcfg->bctl_cnt; ++n) {
 		assert(dcfg->bctl[n] != NULL);
 
+		// If base is already sorted we don't need split it into chunks and sort it.
+		// Instead of that we can just iterate it over in the merge phase.
+		if (datasort_base_is_sorted(dcfg->bctl[n])) {
+			EBLOB_WARNX(dcfg->log, EBLOB_LOG_ERROR, "defrag: datasort_add_view_chunk: start, index %d",
+					n);
+			err = datasort_add_view_chunk(dcfg, n);
+			if (err != 0) {
+				EBLOB_WARNC(dcfg->log, EBLOB_LOG_ERROR, -err, "defrag: datasort_add_view_chunk");
+				goto err;
+			}
+
+			EBLOB_WARNX(dcfg->log, EBLOB_LOG_INFO, "defrag: datasort_add_view_chunk: completed");
+			continue;
```

Comment:
It seems that doing continue here is incorrect because of job with flags eblob_blob_iterate